src/gui/widgets/typed_debugger.cc

```diff
@@ -0,0 +1,967 @@
+            return;
+        }
+        if (equals(nodeType, "char *")) {
+            const bool pointsToString = (memData != nullptr);
+            auto* str = pointsToString ? (char*)memData + startAddress - memBase : nullptr;
+            const auto strLength = pointsToString ? strlen(str) + 1 : 0;
+
+            ImGui::TableNextColumn();  // Size.
+            ImGui::Text("4 (string: %zu)", strLength);
+            ImGui::TableNextColumn();  // Value.
+            if (pointsToString) {
+                ImGui::Text("%s", str);
+                ImGui::SameLine();
+                const auto showMemButtonName =
+                    fmt::format(f_("Show in memory editor##{}{}"), currentAddress, extraImGuiId);
+                if (ImGui::Button(showMemButtonName.c_str())) {
+                    const uint32_t editorAddress = startAddress - memBase;
+                    g_system->m_eventBus->signal(
+                        PCSX::Events::GUI::JumpToMemory{editorAddress, static_cast<unsigned>(strLength)});
+                }
+            }
+            ImGui::TableNextColumn();  // New value.
+            ImGui::TextDisabled("--");
+            ImGui::TableNextColumn();  // Breakpoints.
+            if (watchView) {
+                displayBreakpointOptions(node, currentAddress, memData, memBase);
+            }
+            return;
+        }
+        const auto pointerType = node->type;
+        const auto pointedToType = node->type.substr(0, node->type.size() - 2);
+        node->type = pointedToType;
+        populate(node, m_structs);
+        node->type = pointerType;
+    } else {  // This is a primitive.
+        ImGui::TreeNodeEx(nameColumnString.c_str(), ImGuiTreeNodeFlags_Leaf | ImGuiTreeNodeFlags_Bullet |
+                                                        ImGuiTreeNodeFlags_NoTreePushOnOpen |
+                                                        ImGuiTreeNodeFlags_SpanFullWidth);
+        ImGui::TableNextColumn();  // Type.
+        ImGui::TextUnformatted(node->type.c_str());
+        ImGui::TableNextColumn();  // Size.
+        ImGui::Text("%zu", node->size);
+        ImGui::TableNextColumn();  // Value.
+        const auto basic_offset = startAddress - memBase;
+        uint8_t* memValue = memData + basic_offset;
+        const auto* nodeType = node->type.c_str();
+        printValue(nodeType, memValue);
+        ImGui::TableNextColumn();  // New value.
+        displayNewValueInput(nodeType, memValue);
+        ImGui::TextDisabled("--");
+        ImGui::TableNextColumn();  // Breakpoints.
+        if (watchView) {
+            displayBreakpointOptions(node, currentAddress, memData, memBase);
+        }
+    }
+}
+
+void PCSX::Widgets::TypedDebugger::printValue(const char* type, void* address) {
+    static char s[64];
+    if (equals(type, "char")) {
+        int8_t fieldValue = 0;
+        memcpy(&fieldValue, address, 1);
+        sprintf(s, "%d (0x%2x) \n", fieldValue, fieldValue);
+        ImGui::Text("Value: %s", s);
```

Comment:
Why doing the sprintf to a temporary instead of justusing ImGui::Text's formatting?